k8s/nimbess.yaml

```diff
@@ -0,0 +1,80 @@
+---
+# This ConfigMap is used to configure a self-hosted Nimbess installation.
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: nimbess-config
+  namespace: kube-system
+data:
+  # The CNI network configuration to install on each node.
+  cni_network_config: |-
+    {
+        "name": "k8s-pod-network",
+        "type": "nimbess",
+        "log_level": "info",
+        "grpcServer": "localhost:9111",
+        "ipam": {
+            "type": "whereabouts"
+        }
+    }
+
+---
+
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: nimbess-agent
+  namespace: kube-system
+  labels:
+    k8s-app: nimbess-agent
+spec:
+  selector:
+    matchLabels:
+      k8s-app: nimbess-agent
+  template:
+    metadata:
+      labels:
+        k8s-app: nimbess-agent
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+        scheduler.alpha.kubernetes.io/tolerations: |
+          [{"key": "dedicated", "value": "master", "effect": "NoSchedule" },
+           {"key":"CriticalAddonsOnly", "operator":"Exists"},
+           {"operator":"Exists"}]
+    spec:
+      hostNetwork: true
+      nodeSelector:
+        beta.kubernetes.io/os: "linux"
```

Comment:
does this need to be more specific - like centos doesn't work with BESS ut fedora does. No idea if there is a more specific value either.

Reply:
This really doesn't even need to be there. It was part of me trying to debug why the container wouldn't launch before node was ready, which turned out to be a change in how k8s tolerations are specified. Linux is generic and will work though (kube-proxy uses it)